File: utilities/parser.py
import argparse


def getParser() -> argparse.ArgumentParser:
    """Generates the parser used to parse the command line arguments.

    Returns:
        argparse.ArgumentParser: The parser used to parse the command line arguments.
    """
    parser = argparse.ArgumentParser(prog="quran_cli",
                                     description='Listen to the Quran from your terminal',
                                     epilog='Recall us in your doa\'!',
                                     allow_abbrev=True,
                                     add_help=True,
                                     fromfile_prefix_chars='@')
    # options
    parser.add_argument('-v', action='version', version='%(prog)s 1.0',
                        help='show version')
    parser.add_argument("-i", '--interactive', action='store_true', dest="interactive",
                        help="Run the script in an interactive mode")
    # arguments
    parser.add_argument("-rd", '--radio', action='store', type=int, default='-1', dest='radioIndex', choices=range(0, 148), metavar='[0-147]',
                        help="Specify the radio channel to listen to")
    parser.add_argument("-rc", '--reciter', action='store', type=int, default='-1', dest='reciterIndex', choices=range(0, 247), metavar='[0-246]',
                        help="Specify the reciter to listen to")
    parser.add_argument("-s", '--sura', action='store', type=int, default='-1', dest='suraIndex', choices=range(1, 115), metavar='[1-114]',
                        help="Specify the sura to listen to")
    return parser

File: utilities/parser.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ts_command_line_args_1 = require("ts-command-line-args");
var CommandLine = (function () {
    function CommandLine() {
    }
    CommandLine.getArgs = function () {
        var args = ts_command_line_args_1.parse({
            radio: { type: Number, optional: true, alias: 'd', description: 'Specify the radio channel to listen to' },
            reciterSurah: { type: Number, multiple: true, optional: true, alias: 'c', description: 'Specify the reciter and the surah to listen to. \nIf no surah specified, it will shows the available suras for the specified reciter.' },
            showRadio: { type: Boolean, optional: true, alias: 'n', description: 'Shows all available radio channels' },
            showReciters: { type: Boolean, optional: true, alias: 'r', description: 'Shows all available reciters' },
            showSuras: { type: Boolean, optional: true, alias: 's', description: 'Shows suras numbers' },
            version: { type: Boolean, optional: true, alias: 'v', description: 'Shows the current version' },
            help: { type: Boolean, optional: true, alias: 'h', description: 'Prints this usage guide' },
        }, {
            helpArg: 'help',
            headerContentSections: [{ header: 'Quran-CLI', content: 'Listen to the Quran from your terminal' }],
            footerContentSections: [{ header: '^_^', content: "Recall us in your doa'!" }],
        });
        if (args.version === true) {
            console.log("V1.0.0");
        }
        if (args.reciterSurah == undefined) {
            args.reciterSurah = [];
        }
        return args;
    };
    return CommandLine;
}());
exports.CommandLine = CommandLine;
